code/modules/client/preferences.dm

```diff
@@ -233,6 +233,17 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				randomise_appearance_prefs()
 				save_character()
 
+			// SKYRAT EDIT START - Sanitizing languages
+			var/languages_edited = FALSE
+			for(var/lang_path as anything in languages)
+				var/datum/language/language = new lang_path()
+				if(!(language.type in subtypesof(/datum/language)) || language.secret)
+					languages.Remove(lang_path)
+					languages_edited = TRUE
```

Comment:
Considering how you have this code duplicated in two places, I'd probably recommend just making it a proc that returns true or false based on whether or not languages were edited. Then it'd be a lot cleaner, and a lot easier to maintain in the future.

Reply:
Sounds like a plan!